[Gateway dashboard] Added MQTT connectors config improvements and redesign.

Dmitriymush opened the pull request:
## Pull Request description
![image](https://github.com/thingsboard/thingsboard/assets/54553744/e647786f-c816-4641-8d53-6f555444d48c)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/7c11da4a-d75a-4d61-a2c2-68cf1a130b43)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/26577bd2-3b34-4295-b4f7-6597b97331e4)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/01578064-cc19-4ceb-aef7-40b004feec96)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/1c05a5b5-5397-4cad-b74b-efe5cb410dff)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/450b7ce8-eaa3-4145-bc58-fb933df0eb9b)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/501da3fc-a943-4154-8ce7-319e77d05802)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/f5adf5b7-79df-4e1c-9afa-f5d7a1d00d0f)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/1858fd1a-9e3c-42be-839c-14811c463a4f)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/8765de4b-8f6d-494d-a06d-85e88120897a)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/62322259-61a3-4e4e-9740-945dbbd31933)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/465dfaad-feaa-4a4f-80de-c9e30d55253f)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/40df2b1d-9a20-4706-9d84-e88f8767fd94)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/9599a6f7-aa84-4d27-b1d2-1809f1f17a3c)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/7551ea3e-b938-41cd-9921-d8bd9322fe25)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/a5a486ae-8737-4cf9-8089-41d811f45393)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/d3370933-da09-4760-8a84-fb7481a8f7ea)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/51839271-5bde-420f-989e-a43ace8e6590)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/0c7f341f-ebca-45a0-8110-891a10bcd2f9)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/991d21e0-79b7-4a6f-8c9f-019f452cf099)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/fa5c32fd-28cb-4d9d-bbfb-aac710c301d5)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/481dcaf2-4515-4f33-a577-cf06cb2cb272)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/184b23de-c497-4188-9ed9-bcb09949ec22)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/2a15ce10-efa5-49b6-89b9-da3adae00602)
![image](https://github.com/thingsboard/thingsboard/assets/54553744/230a3d1a-aac3-4bc6-a980-da3385482f0f)


## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [ ] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [ ] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [ ] Changes are backward compatible or upgrade script is provided.
- [ ] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
  
## Front-End feature checklist

- [x] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [x] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [x] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

vvlladd28 commented on ui-ngx/src/app/modules/home/components/widget/lib/gateway/connectors-configuration/mapping-data-keys-panel.component.ts:
```diff
@@ -0,0 +1,175 @@
+  popover: TbPopoverComponent<MappingDataKeysPanelComponent>;
+
+  @Output()
+  keysDataApplied = new EventEmitter<Array<any>>();
```
Need add generic type

vvlladd28 commented on ui-ngx/src/app/modules/home/components/widget/lib/gateway/connectors-configuration/mapping-table.component.ts:
```diff
@@ -0,0 +1,337 @@
+
+}
+
+export class MappingDatasource implements DataSource<any> {
```
Need add generic type

vvlladd28 commented on ui-ngx/src/app/modules/home/components/widget/lib/gateway/dialog/add-connector-dialog.component.html:
```diff
@@ -0,0 +1,102 @@
+      <mat-icon class="material-icons">close</mat-icon>
+    </button>
+  </mat-toolbar>
+  <mat-progress-bar color="warn" mode="indeterminate" *ngIf="isLoading$ | async">
+  </mat-progress-bar>
+  <div style="height: 4px;" *ngIf="!(isLoading$ | async)"></div>
```
Need to change used CSS grid or position absolute

vvlladd28 commented on ui-ngx/src/app/modules/home/components/widget/lib/gateway/dialog/add-connector-dialog.component.html:
```diff
@@ -0,0 +1,102 @@
+  </mat-progress-bar>
+  <div style="height: 4px;" *ngIf="!(isLoading$ | async)"></div>
+  <div mat-dialog-content>
+    <fieldset [disabled]="isLoading$ | async" class="tb-form-panel no-border no-padding" fxLayout="column">
```
Please remove fieldset tags. Need to use function registerDisableOnLoadFormControl

vvlladd28 commented on ui-ngx/src/app/modules/home/components/widget/lib/gateway/dialog/add-connector-dialog.component.scss:
```diff
@@ -0,0 +1,26 @@
+  }
+}
+
+:host ::ng-deep .mat-padding{
+  padding: 0;
+}
```
This is bed practis

vvlladd28 commented on ui-ngx/src/app/modules/home/components/widget/lib/gateway/dialog/mapping-dialog.component.html:
```diff
@@ -0,0 +1,526 @@
+    limitations under the License.
+
+-->
+<form [formGroup]="mappingForm" (ngSubmit)="add()">
```
Maybe for improved delete form tags in top-level dialog

vvlladd28 commented on ui-ngx/src/app/modules/home/components/widget/lib/gateway/gateway-connectors.component.html:
```diff
@@ -22,13 +22,21 @@ <h2>{{ 'gateway.connectors' | translate }}</h2>
       </button>
     </mat-toolbar>
     <div class="table-container">
+      <section *ngIf="!dataSource?.data?.length" fxLayoutAlign="center center"
```
Used is `dataSource.isEmpty()`

Dmitriymush commented:
@vvlladd28 please review carefully commits 
[UI: fixed ordering of tb-hint-tooltip-icon and tb-required elements](https://github.com/thingsboard/thingsboard/pull/10482/commits/c187545917c65a468d00124b8e676479d73a6ad4)
[UI: fixed spacing for tb-required in tb-hint-tooltip-icon](https://github.com/thingsboard/thingsboard/pull/10482/commits/a5b7e686678c134f65bbd301ffc1ce0dc56036c0)
and drop them if that's not the right approach.
